WMIAdventure/backend/WMIAdventure_backend/battle/businesslogic/Outcome.py

```diff
@@ -27,11 +30,34 @@ def is_done(self, turn_num: int) -> bool:
         :param turn_num: Current turn number.
         :return: If battle is completed.
         """
+
+        if self.is_completed:
+            return self.is_completed
+
         if self.attacker.get_hp() <= 0.0 or self.defender.get_hp() <= 0.0 or turn_num > self.MAX_TURNS:
             self.is_completed = True
+            self._calculate_exp_gain()
 
         return self.is_completed
 
+    def _calculate_exp_gain(self):
+        """
+        When battle is finished this method is called to calculate exp
+        gain for each player.
+        """
+
+        winner = self.get_winner()
+
+        if winner is None:
+            self.attacker_exp_gain = draw(self.attacker.level, self.defender.level)
+            self.defender_exp_gain = draw(self.defender.level, self.attacker.level)
+        elif winner is self.attacker:
+            self.attacker_exp_gain = win(self.attacker.level, self.defender.level)
+            self.defender_exp_gain = defeat(self.defender.level, self.attacker.level)
+        elif winner is self.defender:
+            self.attacker_exp_gain = defeat(self.attacker.level, self.defender.level)
+            self.defender_exp_gain = win(self.defender.level, self.attacker.level)
+
```

Comment:
Nie lepiej byłoby to jakoś zamknąć w innym miejscu?
Wyobrażam sobie funkcje, która po prostu dostaje obiekt `Outcome` i na jego podstawie sobie wylicza te kalkulacje. Te `exp_gain` w ogóle nie powinny być w Outcome, tylko właśnie gdzieś poza.